turbopack/crates/turbopack/src/lib.rs

```diff
@@ -140,6 +141,7 @@ async fn apply_module_type(
                     .compile_time_info()
                     .to_resolved()
                     .await?,
+                side_effect_free_packages,
```

Comment:
Other module types could also be flagged as side effect free:

* NodeAddon
* CssModule
* WebAssembly